cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityManager.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.capability;
+
+import com.google.inject.Inject;
+import io.cdap.cdap.api.metadata.MetadataEntity;
+import io.cdap.cdap.api.metadata.MetadataScope;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.metadata.MetadataSearchResponse;
+import io.cdap.cdap.proto.metadata.MetadataSearchResultRecord;
+import io.cdap.cdap.spi.metadata.SearchRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Class with helpful methods for managing capabilities
+ */
+public class CapabilityManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CapabilityManager.class);
+  private static final String CAPABILITY = "capability:%s";
+  private static final String APPLICATION_TAG = "application:%s";
+  private static final String CAPABILITY_KEY = "capability";
+  private static final String APPLICATION = "application";
+  private final MetadataSearchClient metadataClient;
+  private final CConfiguration cConf;
+
+  @Inject
+  CapabilityManager(MetadataSearchClient metadataClient, CConfiguration cConf) {
+    this.metadataClient = metadataClient;
+    this.cConf = cConf;
+  }
+
+  /**
+   * Returns the list of applications that are having metadata tagged with the capability
+   *
+   * @param namespace  Namespace for which applications should be listed
+   * @param capability Capability by which to filter
+   * @param cursor Optional cursor from a previous response
+   * @param offset Offset from where to start
+   * @param limit Limit of records to fetch
+   * @return
+   * @throws Exception - Exception from meta data search if any
+   */
+  public CapabilityApplications getAppsForCapability(NamespaceId namespace, String capability, @Nullable String cursor,
```

Comment:
Just call it `getApplications`. The method is already inside the `CapabilityManager` class, no need to repeat capability in every method.

Reply:
Done